### PR TITLE
Styling for forum plus fixing a few bugs

### PR DESCRIPTION
**Pull request by rmr167**
This PR mostly has styling for forum plus a few bug changes.  ONe big one was changing so that when you like not all descendants are affected.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/ForumServlet.java [hunk 1]
```diff
@@ -93,9 +93,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     if (action.equals(ForumElement.ACCEPTED_PROPERTY)) {
```
Comment (geoffreyzgoog) [inline src/main/java/com/google/sps/servlets/ForumServlet.java:95]: Just noticed now, but these should be "else if".

File: src/main/java/com/google/sps/servlets/UserDataServlet.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
 import java.io.*;
```
Comment (geoffreyzgoog) [inline src/main/java/com/google/sps/servlets/UserDataServlet.java:15]: Just noticed this, but the style guide says no wildcard import.

File: src/main/webapp/scripts/forum-template.js [hunk 3]
```diff
@@ -1,32 +1,42 @@
...
-'<button class="response-button">Post</button>\n' +
-'</div>\n' +
-'<button class="expand-button" style="display: {{repliesDisplay}}">' +
-'See {{numReplies}} Replies</button>\n' +
-'</div>\n' +
-'<div class="replies" id="replies-{{id}}" style="display: none"></div>\n' +
-'<button class="collapse-button" ' +
-'style="display: none">Hide Replies</button>\n' +
+const forumTemplate =
+'<div class="forum-element {{elementType}}">\n' +
```
Comment (geoffreyzgoog) [inline src/main/webapp/scripts/forum-template.js:5]: As Mike mentioned in another PR, you can put the whole template in a single pair of back ticks.

File: src/main/java/com/google/sps/servlets/UserDataServlet.java (hunk 2 above)
Comment (rmr167) [reply src/main/java/com/google/sps/servlets/UserDataServlet.java]: Changed to import java.io.IOException in both UserServlet and ForumServlet

File: src/main/java/com/google/sps/servlets/ForumServlet.java (hunk 1 above)
Comment (rmr167) [reply src/main/java/com/google/sps/servlets/ForumServlet.java]: Updated all in that section to else if

File: src/main/webapp/scripts/forum-template.js (hunk 3 above)
Comment (rmr167) [reply src/main/webapp/scripts/forum-template.js]: Yay! That looks so much better :)

**Pull request merged**